invenio_s3/storage.py

```diff
@@ -17,11 +17,30 @@
 from .helpers import redirect_stream
 
 
+def set_blocksize(f):
+    """Decorator to set the correct block size according to file size."""
+    @wraps(f)
+    def inner(self, *args, **kwargs):
+        size = kwargs.get('size', None)
+        block_size = (
+            size // current_app.config['S3_MAXIMUM_NUMBER_OF_PARTS']  # Integer
+            if size
+            else current_app.config['S3_DEFAULT_BLOCK_SIZE']
+        )
+
+        if block_size > self.block_size:
+            self.block_size = block_size
+        return f(self, *args, **kwargs)
+
+    return inner
+
+
 class S3FSFileStorage(PyFSFileStorage):
     """File system storage using Amazon S3 API for accessing files."""
 
     def __init__(self, fileurl, **kwargs):
         """Storage initialization."""
+        self.block_size = current_app.config['S3_DEFAULT_BLOCK_SIZE']
```

Comment:
Doubt: Is there a specific win of having `block_size` as object attribute instead of returning it (e.g. `@pass_block_size`). It seems it is going to be calculated on every call in any case.

Reply:
If my memory serves me well, the main reason is convenience ☺️ 
The only point where the size is needed is inside `_get_fs` and it doesn't seem right to pass the size from one method to the next every time we call it (which is inside every other method). 
Moreover, I would need to overwrite `save` entirely just to add this parameter when calling `_get_fs`.

WDYT?

Reply:
Makes a lot of sense, didn't catch that part :) Thanks for the answer!